File: CurriculoApp/src/app/curriculo.ts
import { Usuario } from './usuario';

export class Curriculo {
	id: number;
	usuario: Usuario;
	objetivo: string;
    curso: string;
    formacao: string;
    experiencia: string;

	constructor (i: number, u: Usuario, o: string, c: string, f: string, e: string) {
		this.id = i;
		this.usuario = u;
		this.objetivo = o;
		this.curso = c;
		this.formacao = f;
		this.experiencia = e;
	}
}

File: APITrabalhoSAAS/src/main/java/com/trabalhosaas/model/Curriculo.java
package com.trabalhosaas.model;

import javax.persistence.*;

@Entity
public class Curriculo {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private int id;

    @OneToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "id_usuario")
    private Usuario usuario;

    private String objetivo;
    private String curso;
    private String formacao;
    private String experiencia;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public Usuario getUsuario() {
        return usuario;
    }

    public void setUsuario(Usuario usuario) {
        this.usuario = usuario;
    }

    public String getObjetivo() {
        return objetivo;
    }

    public void setObjetivo(String objetivo) {
        this.objetivo = objetivo;
    }

    public String getCurso() {
        return curso;
    }

    public void setCurso(String curso) {
        this.curso = curso;
    }

    public String getFormacao() {
        return formacao;
    }

    public void setFormacao(String formacao) {
        this.formacao = formacao;
    }

    public String getExperiencia() {
        return experiencia;
    }

    public void setExperiencia(String experiencia) {
        this.experiencia = experiencia;
    }
}